06.Lesson_ThinkingProperties/ThinkingPropertiesExercise/MeetingScheduler/meetingscheduler.spec

```diff
@@ -0,0 +1,210 @@
+methods {
+    getStateById(uint256) returns (uint8) envfree
+    getStartTimeById(uint256) returns (uint256) envfree
+    getEndTimeById(uint256) returns (uint256) envfree
+    getNumOfParticipents(uint256) returns (uint256) envfree
+}
+
+definition uninitialized(uint256 meetingId) returns bool = 
+        getStartTimeById(meetingId) == 0 &&
+        getEndTimeById(meetingId) == 0 &&
+        getNumOfParticipents(meetingId) == 0 &&
+        getStateById(meetingId) == 0;
+
+definition initialized(uint256 meetingId) returns bool =
+        getStartTimeById(meetingId) != 0 &&
+        getEndTimeById(meetingId) != 0 &&
+        getStateById(meetingId) != 0;
+
+definition UNINITIALIZED() returns uint256 = 0;
+
+function callFunctionWithParams(method f, uint256 meetingId, uint256 startTime, uint256 endTime) {
+    env e;
+
+    if (f.selector == scheduleMeeting(uint256,uint256,uint256).selector) {
+        scheduleMeeting(e, meetingId, startTime, endTime);
+    } else if (f.selector == startMeeting(uint256).selector) {
+        startMeeting(e, meetingId);
+    } else if (f.selector == cancelMeeting(uint256).selector) {
+        cancelMeeting(e, meetingId);
+    } else if (f.selector == endMeeting(uint256).selector) {
+        endMeeting(e, meetingId);
+    } else if (f.selector == joinMeeting(uint256).selector) {
+        joinMeeting(e, meetingId);
+    } else {
+        calldataarg args;
+        f(e, args);
+    }
+}
+
+// any function called on meeting #1 will have no effect on another meeting
```

Comment:
You have no way of knowing this because we haven't published it yet, but we now have support for natspec-like comments.  I would annotate rules with either `///` or `/**` blocks.